File: code_snippets/ts/src/develop/service_types.ts
import * as restate from "@restatedev/restate-sdk";

const greeterService = restate.object({
    name: "Greeter",
    handlers: {
        greet: async (ctx: restate.ObjectContext, name: string) => {

            // <start_ordering>
            ctx.objectSendClient(Greeter, "Mary").greet( 1);
            ctx.objectSendClient(Greeter, "Mary").greet(2);
            // <end_ordering>

        },
    }
})

const greeterObject = restate.object({
    name: "Greeter",
    handlers: {
        greet: async (ctx: restate.ObjectContext, count: number) => {
            return `Hello ${ctx.key}! ${count}`;
        }
    }
});

export const Greeter: typeof greeterObject = { name : "Greeter" };

File: code_snippets/java/src/main/java/develop/ServiceTypes.java
package develop;

import static counter.CounterOuterClass.*;
import counter.CounterRestate;
import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.common.TerminalException;
import greeter.generated.GreeterProto;
import greeter.generated.GreeterProto.GreetResponse;
import greeter.generated.GreeterRestate;

public class ServiceTypes extends GreeterRestate.GreeterRestateImplBase {

    @Override
    public GreetResponse greet(ObjectContext ctx, GreeterProto.GreetRequest request) throws TerminalException {

        // <start_ordering>
        CounterRestate.CounterRestateClient client = CounterRestate.newClient(ctx);
        client.oneWay().add(
            AddRequest.newBuilder().setName("Restate").setNumber(1).build()
        );
        client.oneWay().add(
            AddRequest.newBuilder().setName("Restate").setNumber(2).build()
        );
        // <end_ordering>

        return GreetResponse.getDefaultInstance();
    }

    private String doDbRequest(){ return ""; }
}